### DISCUSSION TITLE
Allow disabling `rerequestable` when create a check run

### DISCUSSION THREAD

**Discussion by ashi009** [community/community#84929] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

When reporting check results to github, it's perfectly normal that some runs are not re-runable. However, there is no API for setting them, and there is always a `Re-run` button shows next to the failing check run which causes confusion.

However, per the response code of https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#rerequest-a-check-run,

> 403 | Forbidden if the check run is not rerequestable or doesn't belong to the authenticated GitHub App
> -- | --

It seems that there is a field called `rerequestable` associated with check runs, and the same field is also available in the response of https://docs.github.com/en/rest/checks/suites?apiVersion=2022-11-28#get-a-check-suite.

**Comment by Naimulnobel** [comment]
The `rerequestable` field in the GitHub Checks API is a read-only field that indicates whether a check run can be re-run. It's determined by GitHub based on the state of the check run and the permissions of the GitHub App that created it.

Currently, there is no way to manually set a check run as not re-runnable via the API. The "Re-run" button in the GitHub UI is always shown for failed check runs, regardless of whether they are actually re-runnable or not. If a user tries to re-run a check run that is not re-runnable, they will receive a 403 Forbidden error.

This can indeed cause confusion for users, as they might not understand why they are unable to re-run a failed check run. A potential improvement could be to hide the "Re-run" button for check runs that are not re-runnable, or to provide a clearer error message when a user tries to re-run such a check run.

You might want to consider submitting this as a feature request to GitHub. They are always looking for ways to improve their platform and your feedback could help them make it better.

**Comment by almlkbasm** [comment]
.
